File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int, int> sumCount;
        sumCount[0] = 1;
        int count = 0, sum = 0;
        for(int num : nums){
            sum+=num;
            if(sumCount.find(sum - k) != sumCount.end()){
                count += sumCount[sum - k];
            }
            if(sumCount.find(sum) == sumCount.end()){
                sumCount[sum] = 0;
            }
            sumCount[sum]++;
        }
        return count;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # Get counts
        count_map = defaultdict(int)
        for num in nums:
            count_map.setdefault(num,0)
            count_map[num]+=1
        count_arr = [[] for _ in range(len(nums) + 1)]
        for num, count in count_map.items():
            count_arr[count].append(num)
        ans = []
        ctr = 0
        limit_reached = False
        count_arr.reverse()
        for nums in count_arr:
            if not nums:
                continue
            for num in nums:
                ans.append(num)
                ctr+=1
                if ctr == k:
                    limit_reached = True
                    break
            if limit_reached:
                break
        return ans

        